olive/passes/onnx/gptq.py

```diff
@@ -0,0 +1,199 @@
+                description="Calibration dataset. Default value is wikitext2",
+            ),
+            "block_name_to_quantize": PassConfigParam(
+                type_=str,
+                default_value="model.layers",
+                description="Block name to quantize. Default value is model.layers.",
+            ),
+            "group_size": PassConfigParam(
+                type_=int,
+                default_value=128,
+                description="Block size for quantization. Default value is 128.",
+            ),
+            "batch_size": PassConfigParam(
+                type_=int,
+                default_value=1,
+                description="Batch size for quantization. Default value is 1.",
+            ),
+            "seed": PassConfigParam(
+                type_=int,
+                default_value=0,
+                description="Random seed for sampling calibration dataset. Default value is 0.",
+            ),
+            "damp_percent": PassConfigParam(
+                type_=float,
+                default_value=0.01,
+                description="Damping factor for quantization. Default value is 0.01.",
+            ),
+            "static_groups": PassConfigParam(
+                type_=bool,
+                default_value=False,
+                description="Use static groups for quantization. Default value is False.",
+            ),
+            "true_sequential": PassConfigParam(
+                type_=bool,
+                default_value=False,
+                description="Use true sequential for quantization. Default value is False.",
+            ),
+            "desc_act": PassConfigParam(
+                type_=bool,
+                default_value=False,
+                description="Use descriptive activation for quantization. Default value is False.",
+            ),
+            "sym": PassConfigParam(
+                type_=bool,
+                default_value=False,
+                description="Symmetric quantization. Default value is False.",
+            ),
+            "export_optimization": PassConfigParam(
+                type_=str,
+                default_value=None,
+                description="Export optimization level. Default value is None.",
+            ),
+            "data_config": PassConfigParam(
+                type_=Union[DataConfig, Dict],
+                default_value=None,
+                description="""
+                    Data config for quantization. Default value is None.
+                """,
+            ),
+            "dataloader_func": PassConfigParam(
+                type_=Union[Callable, str],
+                default_value=None,
+                description="""Function/function name to generate dataset for quantization.
+                The returned datasets is a list of tokenized data
+                (e.g. [{ 'input_ids': [ 1, 100, 15, ... ],'attention_mask': [ 1, 1, 1, ... ]},...]).
+                Default is None.
+                """,
```

Comment:
@trajepl, I suppose we suggest to use data_config. For new added pass, the dataloader_func is not need supported? I suppose data_config could accomplish similar functionality?

Reply:
I think dataloader_func is useful to pass more custom data without need to meet data_config. Cause sometimes user already have tokenized data.

Reply:
In this PR, it is fine to keep them all.
Could you help add a few TODO comments like:
[TODO](trajepl): consider to use data_config to implement the functionality of dataloader_func. 

I will try to refactor that.